File: backend/app/schemas/schedule_proposal.py
import datetime
import uuid

from pydantic import BaseModel

from app.models.schedule_proposal import ProposalStatus


class ScheduleProposalResponse(BaseModel):
    proposal_id: uuid.UUID
    raw_text_id: uuid.UUID
    title: str
    contents: str
    deadline: datetime.datetime
    status: ProposalStatus
    created_at: datetime.datetime
    updated_at: datetime.datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/scheduleProposal.ts
import type { UserBrief } from './user'

export type ProposalStatus = 'pending' | 'approved' | 'rejected'

export interface ScheduleProposal {
  proposal_id: string
  raw_text_id: string
  title: string
  contents: string
  deadline: string
  status: ProposalStatus
  created_at: string
  updated_at: string
  updated_by: UserBrief | null
  decided_by: UserBrief | null
}
